job_controller/job_controller.go

```diff
@@ -60,6 +61,27 @@ type ControllerInterface interface {
 
 	// Returns the Job from API server
 	GetJobFromAPIClient(namespace, name string) (metav1.Object, error)
+
+	// DeleteJob deletes the job
+	DeleteJob(job interface{}) error
```

Comment:
maybe runtime.Object is better?

Reply:
And CreateJob missing?

Reply:
Some methods require both "metav1.Object" (for things like GetName,GetNamespace) and "runtime.Object" (for ".Recorder.Eventf")types, so thought it's better to be make it consistent to be interface{}

Reply:
SGTM

Reply:
Yeah, same question with @johnugeorge 

Can we define CreateJob here?

Reply:
CreateJob is actually done by the user when creating a CRD (i.e. TFJob). I think operators will not create its own CRDs ?

Reply:
Yeah, sure, I think so, thanks.